### PR TITLE
Hotfix: refactor tech docs pipeline to match dev docs

### PR DESCRIPTION
**Pull request by phantas**
Currently we're dealing with potential secrets leakage management...
The way we're deploying things is printing out the credentials to the
pipeline. This isn't the worst thing ever, as Concourse is only
accessible by people within the Bardeen network.

This however, is still potential threat... It could lead to GovWifi
product page being overwritten with no audit of when, why or who...

This makes the pipeline at least consistent with other pipelines

### CODE REVIEW

**Comment by steventux** [comment]
@phantas what do you advise we do with this PR? The build is unstable or stalled, I'm happy to approve in ignorance.

File: ci/pipelines/build-and-deploy.yml [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+    - get: govwifi-tech-docs
+      trigger: true
+    - task: set-pipelines
+      file: tech-ops/ci/tasks/self-updating-pipeline.yaml
+      input_mapping: {repository: govwifi-tech-docs}
+      params:
+        CONCOURSE_TEAM: govwifi
+        CONCOURSE_PASSWORD: ((readonly_local_user_password))
+        PIPELINE_PATH: ci/pipelines/build-and-deploy.yml
+        PIPELINE_NAME: tech-docs
```
Comment (paroxp) [inline ci/pipelines/build-and-deploy.yml]:
```suggestion
        PIPELINE_NAME: tech-docs-deploy
```

Current name.

File: ci/pipelines/build-and-deploy.yml [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+          inputs:
+            - name: govwifi-tech-docs
+              path: repo
+          outputs:
+            - name: bundled
+          run:
+            path: sh
+            dir: repo
+            args:
+            - -c
```
Comment (paroxp) [inline ci/pipelines/build-and-deploy.yml]:
```suggestion
            - -eu
            - -c
```

File: ci/pipelines/build-and-deploy.yml [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+            - |
+              apt-get update
+              apt-get install -y nodejs
```
Comment (paroxp) [inline ci/pipelines/build-and-deploy.yml]: Things like the two above will add huge amount of time to the execution of the pipeline. Would you rather use a prebuilt docker image that includes all the tools you need?

File: ci/pipelines/build-and-deploy.yml [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+  - name: tech-ops
+    type: git
+    source:
+      uri: https://github.com/alphagov/tech-ops.git
+
+  - name: govwifi-tech-docs
+    type: git
+    source:
+      uri: "https://github.com/alphagov/govwifi-tech-docs.git"
+      branch: master
```
Comment (paroxp) [inline ci/pipelines/build-and-deploy.yml]:
```suggestion
```

File: ci/pipelines/build-and-deploy.yml [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+
+  - name: govwifi-tech-docs-pr
+    type: pull-request
+    check_every: 1m
```
Comment (paroxp) [inline ci/pipelines/build-and-deploy.yml]:
```suggestion
```

File: ci/pipelines/build-and-deploy.yml [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+  - name: deploy-to-paas
+    # See https://github.com/alphagov/tech-ops/blob/master/reliability-engineering/pipelines/internal-apps.yml
+    type: cf
+    source:
+      api: "https://api.cloud.service.gov.uk"
+      organization: "govwifi"
+      space: "production"
+      username: ((govpaas-username))
+      password: ((govpaas-password))
+      skip_cert_check: false
```
Comment (paroxp) [inline ci/pipelines/build-and-deploy.yml]:
```suggestion
```

? 😱

File: ci/pipelines/build-and-deploy.yml [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+            - name: govwifi-tech-docs-pr
+              path: repo
+          run:
+            path: sh
+            dir: repo
+            args:
+            - -c
+            - |
+              apt-get update
+              apt-get install -y nodejs
```
Comment (paroxp) [inline ci/pipelines/build-and-deploy.yml]: See comment below.

File: ci/pipelines/build-and-deploy.yml (hunk 3 above)
Comment (phantas) [reply ci/pipelines/build-and-deploy.yml]: eventually, I'll add a card. Right now this PR is stale enough as is.

**Pull request merged**